Cython/Compiler/ParseTreeTransforms.py

```diff
@@ -913,6 +913,8 @@ def visit_SingleAssignmentNode(self, node):
         return node
 
     def visit_NameNode(self, node):
+        if node.annotation:
+            node.annotation = self.visit(node.annotation)
```

Comment:
I don't think we're re-assigning the result of `self.visit(...)` anywhere in this transform. And `visit_AnnotationNode()` also doesn't modify the node result.
```suggestion
            self.visit(node.annotation)
```
That being said, I think there should be a method `visitchild(self, node, str name)` that getattrs the child node, visits it, and then reassigns the attribute if it changed. That would be the clean way. I'll add that.